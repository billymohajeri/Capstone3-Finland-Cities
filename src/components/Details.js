import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchCitiesDetails } from '../redux/cityDetailsSlice';
import '../styles/Details.css';

const Details = () => {
  const citiesDetailsList = useSelector((state) => state.details.citiesDetails);
  const dispatch = useDispatch();

  const { id } = useParams();
  useEffect(() => {
    dispatch(fetchCitiesDetails(id));
  }, [dispatch, id]);

  return (
    <div className="details-container">
      {citiesDetailsList.data && (
        <>
          <div>
            <h2 className="text-city">
              City:&nbsp;
              {citiesDetailsList.data.city}
              <img
                src={`https://7gg9.short.gy/COA${citiesDetailsList.data.id}`}
                alt={citiesDetailsList.data.city}
              />
            </h2>
          </div>
          <ul className="details-list">
            <li>
              <p className="info-text">
                Region:&nbsp;
                {citiesDetailsList.data.region}
              </p>
            </li>
            <li>
              <p className="info-text">
                Latitude:&nbsp;
                {citiesDetailsList.data.latitude}
              </p>
            </li>
            <li>
              <p className="info-text">
                Longitude:&nbsp;
                {citiesDetailsList.data.longitude}
              </p>
            </li>
            <li>
              <p className="info-text">
                Population:&nbsp;
                {citiesDetailsList.data.population}
              </p>
            </li>
            <li>
              <p className="info-text">
                Elevation Meters:&nbsp;
                {citiesDetailsList.data.elevationMeters
                  ? citiesDetailsList.data.elevationMeters
                  : 'Not provided...'}
              </p>
            </li>
            <li>
              <p className="info-text">
                Time Zone: UTC+2&nbsp;
                {citiesDetailsList.data.timezone}
              </p>
            </li>
          </ul>
        </>
      )}
    </div>
  );
};

export default Details;
