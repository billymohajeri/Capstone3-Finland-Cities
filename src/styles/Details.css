body {
  margin: 0;
}

.details-container {
  background-color: #9b325a;
  border: 1px solid black;
}
