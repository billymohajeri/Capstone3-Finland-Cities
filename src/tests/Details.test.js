import React from 'react';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import Details from '../components/Details';
import citiesDetailsSlice from '../redux/cityDetailsSlice';

const createTestStore = () => {
  const store = configureStore({
    reducer: {
      details: citiesDetailsSlice,
    },
  });

  return store;
};
export default createTestStore;

describe('Connected React/Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = createTestStore();

    component = renderer
      .create(
        <Provider store={store}>
          <Details />
        </Provider>,
      )
      .toJSON();
  });

  test('Renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
