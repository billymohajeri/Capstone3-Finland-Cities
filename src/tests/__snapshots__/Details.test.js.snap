// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected React/Redux Component Renders correctly 1`] = `
<div
  className="first-container"
/>
`;
